### PR TITLE
Sync parameter events callbacks

### PR DESCRIPTION
**Pull request by esteve**
Adds examples for `on_parameter_event`

Connects to ros2/rclcpp#49

### CODE REVIEW

File: rclcpp_examples/src/parameters/set_and_get_parameters.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ int main(int argc, char ** argv)
 
   auto parameters_client = std::make_shared<rclcpp::parameter_client::SyncParametersClient>(node);
 
-  // Set several differnet types of parameters.
+  // Set several different types of parameters.
```
Comment (wjwwood) [inline rclcpp_examples/src/parameters/set_and_get_parameters.cpp]: Doh.

**Comment by wjwwood** [comment]
lgtm

File: rclcpp_examples/src/parameters/set_and_get_parameters.cpp (hunk 1 above)
Comment (esteve) [inline rclcpp_examples/src/parameters/set_and_get_parameters.cpp]: I could be wrong, though. Maybe there's the internet, the samenet and the differnet :stuck_out_tongue_winking_eye:

**Comment by tfoote** [comment]
+1

**Comment by dirk-thomas** [comment]
lgtm

**Comment by jacquelinekay** [comment]
looks fine. At some point we should consider adding a lot more comments to all the examples, maybe giving a summary of what the example does at the top and a little bit of motivation about why it is exists. But this could be done at a later point, since the selection of examples is still evolving.

**Pull request merged**